contracts/contracts/processors/BaseProcessor.sol

```diff
@@ -68,4 +68,29 @@ contract BaseProcessor is Ownable {
         require(!nullifierRegistry.isNullified(_nullifier), "Nullifier has already been used");
         nullifierRegistry.addNullifier(_nullifier);
     }
+
+    // stringToUint is used to convert a string like "45" to a uint256 4
+    function _stringToUint(string memory s, uint256 expectedDecimals) internal pure returns (uint256) {
```

Comment:
We should mock this function and add test cases for all the 3 cases of `expectedDecimals` and `decimalPlaces` (lt, equals, gt).